File: 14-longest-common-prefix/longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        l = len(strs)
        if l == 0:
            return ""
        
        if l == 1:
            return strs[0]

        n = len(strs[0])
        for i in range(1, l):
            x = len(strs[i])
            if x < n:
                n = x

        for i in range(n):
            ch = strs[0][i]
            for j in range(l):
                if strs[j][i] != ch:
                    return strs[0][:i]
        
        return strs[0][:n]

File: 14-longest-common-prefix/longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if(strs.empty()) return "";

        sort(strs.begin(), strs.end());

        string first = strs[0];
        string last = strs[strs.size() - 1];
        string pref = "";

        int minLen = min(first.size(), last.size());

        for(int i = 0; i < minLen; i++) {
            if(first[i] != last[i]) break;
            pref += first[i];
        }

        return pref;
    }
};